supervisor/core/instance.go

```diff
@@ -55,8 +58,8 @@ func (inst *Instance) Start() error {
 	return inst.Cmd.Start()
 }
 
-// Restart restarts the Instance.
-func (inst *Instance) Restart() error {
+// RestartInstance restarts the Instance.
+func (inst *Instance) RestartInstance() error {
```

Comment:
This change looks bad. Maybe `inst.Restart()` and `inst.ShouldBeRestartedOnFail`?

Reply:
`OmgPleaseNo` I hate myself for `RestartAfterTermInstance`. I rename it to `Restartable`.